frontend/apps/filemanager/filemanagerbookinfo.lua

```diff
@@ -173,17 +175,17 @@ function BookInfo:show(file, book_props)
 end
 
 function BookInfo.getCustomProp(prop_key, filepath)
-    local custom_metadata_file = DocSettings:getCustomMetadataFile(filepath)
+    local custom_metadata_file = DocSettings:findCustomMetadataFile(filepath)
     return custom_metadata_file
-        and DocSettings:openCustomMetadata(custom_metadata_file):readSetting("custom_props")[prop_key]
+        and DocSettings.openSidecarFile(custom_metadata_file):readSetting("custom_props")[prop_key]
 end
 
 -- Returns extended and customized metadata.
 function BookInfo.extendProps(original_props, filepath)
     -- do not customize if filepath is not passed (eg from covermenu)
-    local custom_metadata_file = filepath and DocSettings:getCustomMetadataFile(filepath)
+    local custom_metadata_file = filepath and DocSettings:findCustomMetadataFile(filepath)
     local custom_props = custom_metadata_file
-        and DocSettings:openCustomMetadata(custom_metadata_file):readSetting("custom_props") or {}
+        and DocSettings.openSidecarFile(custom_metadata_file):readSetting("custom_props") or {}
```

Comment:
About the namings (just mentionning that, up to you if we can find better names, not a real issue).
Looks like openSidecarFile() can be used to open any kind of Lua dump file (docsettings, custom metadata) in the sidecar directory.
Looks a bit odd, but I was ok with that.
Until I saw in docsettings.lua that findSidecarFile and getSidecarFilename (so, also mentionnign "SidecarFile") do work only on the docsetting (ie. metadata.epub.lua).
So, dunno if you should keep "SidecarFile" for any file - and use "DocSettingsFile" (or another name) for metadata.epub.lua ?
Or just rename openSidecarFile() to openFileInSidecarDir() ?

Reply:
With minor changes in it, we can just use:
https://github.com/koreader/koreader/blob/f3520effd679eb24a352e7dec600c5b378c3d376/frontend/luasettings.lua#L21

Reply:
instead of what?
Feels like this :open() is a high level public function. So it might be best to keep a low-level openFileInSidecarDir() distinct.

Reply:
Maybe `openSettingsFile`? Ready for `readSetting`. Can be in any dir.

Reply:
Fine with that.
(Although reading "openSettingsFile", I would expect it open docsettings, which it can open any Lua dump-file like your custom metadata file, which is not really a "setting" as in docsetting :) but just a Lua dump-file. But  openLuaDataFile sounds odd to :/ so ok with openSettingsFile() if you feels it's best.)